### PR TITLE
Modified clock design, added new time source

### PR DESCRIPTION
**Pull request by patrykantosz**
## What does this PR do?

Changes the current design of the Clock source and adds new time source - the ROS 2 frame. Now there is only one Clock, `ROS2Clock`, which can be configured to work with different time sources. The Clock can also publish time to the `/clock` topic - this option is enabled by default. There are three TimeSources:
- realtime - it starts on the start of the simulation (or on the start of the application, when you use Editor). Can be affected by the jitter in the data, simulation computations or other similar events.
- simulation - provides timestamps calculated using updates from the physics engine. Simulation paces the clock with stable steps, and as a result, this time source eliminates data jitter. This is a default time source.
- ros2 - provides the time taken from the ROS 2. By default this is system_time but it can be configured to provide the time from the other sources.

Clock can be configured via `.setreg` file or by passing `--setreg` flags in the command line. Example:
` ./build/linux_working/bin/profile/Editor --regset="/O3DE/ROS2/ClockType=ros2" --regset="/O3DE/ROS2/PublishClock=true"`

## How was this PR tested?

By checking if ROS 2 `/clock` topic is available and checking timestamps there for each of the time sources.

### CODE REVIEW

**Review by pijaro — APPROVED** [review]
LGTM

**Review by michalpelka — APPROVED** [review]
Happy with change.

File: Gems/ROS2/Code/Source/SystemComponents/ROS2SystemComponent.cpp [hunk 1]
```diff
@@ -113,21 +116,38 @@ namespace ROS2
...
             {
-                AZ_Printf("ROS2SystemComponent", "Enabling Physical steady clock");
-                m_simulationClock = AZStd::make_unique<PhysicallyStableClock>();
+                m_simulationClock = AZStd::make_unique<ROS2Clock>(clocksMap[clockType](), publishClock);
                 return;
             }
         }
-        AZ_Printf("ROS2SystemComponent", "Enabling realtime clock");
-        m_simulationClock = AZStd::make_unique<SimulationClock>();
+        AZ_Info("ROS2SystemComponent", "Cannot read registry or the clock type is unknown, enabling simulation clock.");
```
Comment (spham-amzn) [inline Gems/ROS2/Code/Source/SystemComponents/ROS2SystemComponent.cpp:149]:
```suggestion
        AZ_Info("ROS2SystemComponent", "Cannot read registry or the clock type '%s' is unknown, enabling simulation clock.", clockType.c_str());
```

File: Gems/ROS2/docs/guides/ros2-gem.md [hunk 2]
```diff
@@ -194,6 +199,21 @@ All used services types are defined in gazebo_msgs package.
 - Detailed spawn point info access: spawn point name should be passed in request.model_name. Defined pose is sent in response.pose.
   - example call: `ros2 service call /get_spawn_point_info gazebo_msgs/srv/GetModelState '{model_name: 'spawn_spot'}'`
 
+## Clock
+You can control how simulation time is handled, which affects timestamps on all the data coming from the simulation. `ROS2Clock` class captures this behavior and publishes, or not, current time to `\clock` topic based on a configuration setting .
```
Comment (spham-amzn) [inline Gems/ROS2/docs/guides/ros2-gem.md:203]:
```suggestion
You can control how simulation time is handled, which affects timestamps on all the data coming from the simulation. `ROS2Clock` class captures this behavior and publishes, or not, current time to `\clock` topic based on a configuration setting.
```

**Review by spham-amzn — APPROVED** [review]

**Comment by amzn-changml** [comment]
This has passed AR, but the status is stuck on an older review: https://jenkins.build.o3de.org/blue/organizations/jenkins/o3de-extras/detail/PR-707/5/pipeline. Moving to override now.

**Pull request merged**